From f1c0a19f23e06a71bda72757facae7a6a264d525 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 10 Nov 2019 13:55:50 +0200
Subject: [PATCH] SW-6739] Makefile: update for LTO issue

needed to support LTO properly by passing proper --plugin flag to ar and
ranlib

Change-Id: Ic583eb8470565d03ada31ccd1e4e9b354464db4e
---

diff --git a/Makefile b/Makefile
index fad35bf..74ba1cc 100644
--- a/Makefile
+++ b/Makefile
@@ -2,8 +2,9 @@
 AR = $(CROSS_COMPILE)ar
 RANLIB = $(CROSS_COMPILE)ranlib
 ifneq ($(CROSS_COMPILE),)
-	LTO_PLUGIN := $(shell find $(abspath ../../) -name liblto_plugin.so)
-	RANLIB += --plugin=$(LTO_PLUGIN)
+	export LTO_PLUGIN := --plugin $(shell find $(abspath ../../) -name liblto_plugin.so)
+	RANLIB += $(LTO_PLUGIN)
+	export ARFLAGS += $(LTO_PLUGIN)
 endif
 export CC AR RANLIB
 OBJCOPY = $(CROSS_COMPILE)objcopy
@@ -14,7 +15,7 @@
 CFLAGS += -Isdk/src/include
 CFLAGS += -Isdk/elpsoft/src/headers
 CFLAGS += -Isdk/elpsoft/src/math
-#LDFLAGS = -flto
+LDFLAGS = -flto
 LDFLAGS += -L $(LIBDIR) -lsb -lellipsys
 LDFLAGS += -Wl,--defsym=lf2flt=0x124
 
@@ -22,9 +23,10 @@
 all: arcboot.bin arcboot_rom.bin
 	@touch arcboot.bin
 
-libsb libellipsys sbtool:
+libsb libellipsys:
+	@$(MAKE) -C sdk $@ SB_SIGN_MODE=rsa_std
+sbtool:
 	@$(MAKE) -C sdk $@
-
 %.bin: %
 	@$(OBJCOPY) -O binary -R .note.gnu.build-id $< $@
 
